src/content/docs/en/guides/upgrade-to/v3.mdx

```diff
@@ -0,0 +1,614 @@
+---
+title: Upgrade to Astro v3
+description: How to upgrade your project to the latest version of Astro.
+i18nReady: false
+---
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+import FileTree from '~/components/FileTree.astro'
+
+
+This guide will help you migrate from Astro v2 to Astro v3.
+
+Need to upgrade an older project to v2? See our [older migration guide](/en/guides/upgrade-to/v2/).
+
+## Upgrade Astro
+
+Update your project's version of Astro to the latest version using your package manager. If you're using Astro integrations, please also update those to the latest version.
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  # Upgrade to Astro v3.x
+  npm install astro@latest
+  
+  # Example: upgrade React and Tailwind integrations
+  npm install @astrojs/react@latest @astrojs/tailwind@latest
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  # Upgrade to Astro v3.x
+  pnpm install astro@latest
+
+  # Example: upgrade React and Tailwind integrations
+  pnpm install @astrojs/react@latest @astrojs/tailwind@latest
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  # Upgrade to Astro v3.x
+  yarn add astro@latest
+  
+  # Example: upgrade React and Tailwind integrations
+  yarn add @astrojs/react@latest @astrojs/tailwind@latest
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+## Astro v3.0 Experimental Flags Removed
+
+Remove the following experimental flags from `astro.config.mjs`:
+
+```js del={5-8}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  experimental: {
+    assets: true,
+    viewTransitions: true,
+  },
+})
+```
+
+These features are now available by default:
+
+- [View Transitions](/en/guides/view-transitions/) for animated page transitions and persistent islands.
+- A new image services API `astro:assets` for [using images in Astro](/en/guides/images/), including a new `<Image />` component and `getImage()` function.
```

Comment:
```suggestion
- A new image services API `astro:assets` for [using images in Astro](/en/guides/images/), including a new `<Image />` component and `getImage()` function. Please read the detailed [image upgrade advice](/en/guides/images/#upgrade-to-v30-from-v2x) to see how this might affect your project.
```